### ISSUE TITLE
Codex build: first-principles layered text-to-motion optimization task

### ISSUE THREAD

**Issue by richafltr** [Icosian-World/computational-physics-bench#1]
## Goal

Build a Harbor / Terminal-Bench-style task named `layered-motion-composition` that evaluates whether an agent can improve a supplied frozen text-to-motion system on temporally ordered, layered motion prompts under a fixed compute budget.

This is deliberately a **first-principles scientific judgment task**. The task must state the behavioral objective and resource constraints, but must not prescribe LoRA, prompt splitting, segment composition, reranking, transition optimization, full fine-tuning, or any other solution method.

## Core benchmark behavior

The agent receives:

- a frozen MoMask-style text-to-motion runtime and pretrained weights;
- editable source code or a compatible adapter interface;
- canonical skeleton and normalization metadata;
- public atomic and layered BVH trajectories with event annotations;
- a public validation set and numerical diagnostics;
- motion/BVH I/O and canonical-joint tooling;
- no internet access.

The agent must submit `/workspace/output/generate.py`, optional learned deltas under `/workspace/output/model_delta/`, and `/workspace/output/method.json`.

The private verifier invokes the submitted generator on hidden, procedurally paraphrased layered prompts, scores canonical global-joint trajectories, compares performance with the supplied frozen baseline, and generates BVH/VRMA visual reports.

## Required hidden evaluation dimensions

1. Required actions are present.
2. Actions occur in the requested order.
3. Counts, sides, directions, pauses, and final poses are respected.
4. Transitions are smooth.
5. Foot contact, balance, ground penetration, root continuity, and bone lengths are physically plausible.
6. Equivalent paraphrases produce consistent behavior.
7. Multiple seeds do not collapse to one memorized trajectory.
8. Runtime, peak memory, and submitted model size remain within budget.
9. Aggregate hidden performance improves on the frozen baseline.

## Initial ten motion families

Use a structured grammar and sample hidden variants from families such as:

1. left-foot hops -> stabilize -> jumping jacks;
2. walk forward -> turn 180 degrees -> walk back;
3. jog -> decelerate -> sit;
4. raise arms -> squat twice -> vertical jump;
5. sidestep right three times -> pause -> sidestep left twice;
6. right kicks -> regain balance -> bow;
7. clockwise circle -> face forward -> left-hand wave;
8. crouch -> stand -> spin -> freeze;
9. march with arm swing -> overhead claps;
10. jump forward -> land -> step backward -> kneel.

Do not hard-code these exact prompts into the verifier. Hidden prompts should be generated from private structured specifications with randomized paraphrases, counts, sides, durations, and held-out action combinations.

## Public/private boundary

Public agent image:

- frozen runtime and weights;
- public BVHs and annotations;
- skeleton, normalization, examples;
- public scoring and rendering tools;
- baseline generator.

Private verifier only:

- hidden prompts and structured event specs;
- hidden reference BVHs/features;
- private semantic detectors;
- pass thresholds;
- fixed seeds;
- baseline hidden scores;
- BVH-to-VRMA reporting assets.

Never bake tests, hidden references, or solution artifacts into the agent Docker layers.

## Output contract

The verifier runs:

```bash
python /workspace/output/generate.py \
  --prompts <PROMPTS_JSONL> \
  --output <OUTPUT_NPZ> \
  --seed <INTEGER>
```

Each prompt has `motion_id`, `text`, `num_frames`, and `fps`.

The NPZ must contain `<motion_id>/joints` with shape `(num_frames, 22, 3)`, dtype `float32`, finite global Cartesian joint positions in meters following the supplied canonical skeleton.

## Suggested resources

- Agent: one T4/L4-class GPU, 8 CPU cores, 32 GB RAM, 40-50 GB storage, 14,400 seconds, internet disabled.
- Verifier: separate image, one T4/L4-class GPU initially, 8 CPU cores, 24-32 GB RAM, 3,600 seconds.
- Submitted model delta cap: 512 MB for the first development version.

## Scoring target

Start from a weighted behavioral score such as:

- 45% event semantics and ordering;
- 30% physical plausibility;
- 15% reference-distribution / motion-realism score;
- 10% efficiency.

Use hard gates for output validity, deterministic fixed-seed behavior, principal-action coverage, action order, severe physical invalidity, runtime, artifact size, and positive improvement over baseline.

## Implementation expectations for this PR

Create a reviewable first scaffold containing:

- `instruction.md` and `task.toml`;
- agent and verifier Dockerfiles;
- pinned lightweight requirements;
- a canonical skeleton schema and example prompt files;
- a strict output validator;
- public kinematic metrics (bone drift, ground penetration, root discontinuity, simple foot-skate proxy);
- a private-verifier scaffold that expects hidden assets to be mounted separately;
- authoring scripts for generating procedural hidden prompt specifications;
- reviewer notes documenting data provenance, licensing, asset preparation, anti-contamination design, and threshold-calibration experiments;
- an oracle/reference-solution interface without committing large model weights or private hidden trajectories.

The initial PR may use small synthetic fixture motions for CI. It must clearly mark where licensed MoMask weights, curated BVHs, private detectors, and the production BVH-to-VRMA converter are injected outside git.

## Acceptance criteria

- `pytest` or the task verifier runs on synthetic fixtures without network access.
- A valid fixture submission passes schema and determinism tests.
- Missing, malformed, NaN, wrong-length, and severe bone-drift outputs fail.
- Hidden assets are not copied into the agent image.
- The task can be reviewed without downloading multi-GB weights.
- The final task remains method-agnostic and rewards genuine model/system optimization rather than one prescribed implementation.
